smartgymapi/handlers/user.py

```diff
@@ -84,3 +100,138 @@ def delete(self):
+
+            if len(recommended_buddies) < 5:
+                # List not full, we can continue to the next iteration early
+                recommended_buddies[user] = favorite_weekday_similarity
+                continue
+
+            # Get the lowest similarity currently in the list
+            lowest_user = min(recommended_buddies,
+                              key=recommended_buddies.get)
+            lowest_similarity = recommended_buddies[lowest_user]
+
+            # Replace the lowest user if this user fits the current user
+            # better.
+            if favorite_weekday_similarity > lowest_similarity:
+                recommended_buddies.pop(lowest_user, None)
+                recommended_buddies[user] = favorite_weekday_similarity
+
+        recommended_buddies_data = UserSchema(many=True).dump(
+            recommended_buddies).data
+        RedisSession().session.set(
+            self.redis_key,
+            recommended_buddies_data)
+
+        return recommended_buddies_data
+
+    @view_config(context=BuddyFactory, request_method="PUT")
+    def put(self):
+        schema = BuddySchema(strict=True)
+        try:
+            result, errors = schema.load(self.request.json_body)
+            schema.validate_user_id(result, self.request.user.id)
+        except ValidationError as e:
+            raise HTTPBadRequest(json={'message': str(e)})
+
+        new_buddy = get_user(result['user_id'])
+        new_buddy_id = str(new_buddy.id)
+
+        # add the new buddy to the user's existing buddies
+        self.request.user.self_to_buddies.append(new_buddy)
+
+        try:
+            persist(self.request.user)
+            response_body = UserSchema().dump(new_buddy).data
+        except:
+            log.critical("Something went wrong adding a new buddy",
+                         exc_info=True)
+            rollback()
+        finally:
+            commit()
+
+        # If the new buddy was a recommended one we have to search for a new
+        # recommendatation
+        cached_recommendations = RedisSession().session.get(self.redis_key)
+        if (cached_recommendations and
+                new_buddy_id in cached_recommendations.decode('utf-8')):
+            # Invalidate the cached recommended buddies
+            RedisSession().session.delete(self.redis_key)
+
+        return response_body
+
+    @view_config(context=User, request_method="DELETE")
+    def delete(self):
+        try:
+            self.request.user.buddies_to_self.remove(self.request.context)
+        except ValueError:
+            # It's possible that the relationship only existed one way
+            # which means one will raise a key error
+            pass
+
+        try:
+            self.request.user.self_to_buddies.remove(self.request.context)
+        except ValueError:
+            # It's possible that the relationship only existed one way
+            # which means one will raise a key error
+            pass
```

Comment:
Ja alleen het probleem was dat als ik het in 1 try zette dat die als de eerste methode een ValueError gaf de tweede methode niet meer uitvoerde